### DISCUSSION TITLE
Question: assign prior distribution using for loop in pymc

### DISCUSSION THREAD

**Discussion by zjuhxy** [pymc-devs/pymc3#6898] [General]
I'm writing a hierarchical model using pymc and have a little problem. I want to implement the following: a total of N people are dividing into 2 groups. The group xi ~ Bernoulli(0.5). for person i, if xi=0 (belonging to the first group), yi ~ N(mu1, sigma1); else if xi=1(belonging to the second group), yi~N(mu2,sigma2). Here is my code

```
with pm.Model():
    xi = pm.Bernoulli('xi', 0.5, size=N)
    for i in range(N):
        if xi[i] == 0:
            yi[i] = pm.Normal(mu1,sigma1)
        else:
            yi[i] = pm.Normal(mu2,sigma2)
```

This triggered a typeError: 'TensorVariable' object does not support item assignment.

This is actually a small part of my model. So how can I implement a similar function, i.e., how can I assign values to certain prior distribution separately for a larger group of subjects?

**Comment by fonnesbeck** [comment]
Please post usage-related questions to our [Discourse page](https://discourse.pymc.io/). The discussions here are intended for development-related topics.
